core/templates/dev/head/forms/formBuilder.js

```diff
@@ -598,9 +603,22 @@ oppia.factory('rteHelperService', [
       // Returns a DOM node.
       createRteElement: function(componentDefn, customizationArgsDict) {
         var el = $('<img/>');
-        el.attr('src', componentDefn.iconDataUrl);
+        if (componentDefn.previewUrlTemplate) {
+          el.attr('src',
+            $interpolate(
+              componentDefn.previewUrlTemplate)(
+                angular.extend(customizationArgsDict,
+                  {
+                    explorationId: explorationContextService.getExplorationId()
```

Comment:
I worry a bit about this, since it's conceivable we may use RTEs outside the context of an exploration (e.g. in the admin page).

Did you check that explorationContextService is available in all pages (so we don't get a dependency error)?
Also, this means that, if we aren't in an exploration-context page, we should probably ban the image component.